internal/telemetry/utils.go

```diff
@@ -0,0 +1,111 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+// Package telemetry implements a client for sending telemetry information to
+// Datadog regarding usage of an APM library such as tracing or profiling.
+package telemetry
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"sync"
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/httpmem"
+)
+
+// TestHTTPClient provides a test http client that waits for a specific sequence of telemetry events.
+// It returns a function that waits for all expected events to be received, as well as a cleanup function.
+func TestHTTPClient(ctx context.Context, t *testing.T, events []RequestType, ignoreEvents []RequestType) (client *http.Client, wait func() []*Body, cleanup func()) {
```

Comment:
Let's create a package called `internal/telemetry/telemetrytest` and pop it in there instead. Then you can just call this `HttpClient` since it'll be apparent that it's for testing from the package name.